### DISCUSSION TITLE
List the queues and bindings declared by a given connection

### DISCUSSION THREAD

**Discussion by metalalive** [gftea/amqprs#132] [General]
Hi, is there any public API which lists all queues and bindings declared by a given [connection](https://docs.rs/amqprs/1.6.1/amqprs/connection/index.html) instance ? For example, in crate lapin , there is [`Connection::topology()`](https://docs.rs/lapin/latest/lapin/struct.Connection.html#method.topology)
 
I have an application which declares several queues to use as soon as a new connection has been established, and prefers to do this exactly once.

I search the doc (by keywords such as `declare`, `topo`) but have not found any relevant  function / method

Appreciate any feedback , thanks

**Comment by gftea** [comment]
Hi, such information is available from rabbitmq server.

**Reply by metalalive** [reply]
Thanks for the reply , could you please give me a hint / keyword about how to retrieve that information ?
I am reading [AMQP v0.9.1 reference](https://www.rabbitmq.com/amqp-0-9-1-reference) and cannot find out anything relevant,
Hopefully I don't misunderstand , thanks again

**Reply by gftea** [reply]
See this if meeting your needs https://www.rabbitmq.com/docs/management#http-api

**Reply by metalalive** [reply]
Thank you
